Add examples summary page (#4354)
expressjs/express: examples/README.md

# Express examples

This page contains list of examples using Express.

> - Authentication with login and passowrd - [examples/auth](./auth)

Reviewer: ```suggestion
- Authentication with login and password - [examples/auth](./auth)
```

typo

- HTTP content negotiation - [examples/content-negotiation](./content-negotiation)
- Working with cookie-based session - [examples/cookie-sessions](./cookie-sessions)

> - Working with cookies via [`cookie-parser`](https://github.com/expressjs/cookie-parser) - [examples/cookies](./cookies)

Reviewer: This is just a generic cookie example, nothing specific with cookie-parser. It is to demo the res.cookie and res.clearCookie APIs in Express

Author: I've changed it to just `Working with cookies`

> - Files transfer to client - [examples/downloads](./downloads)

Reviewer: The description of this one sounds strange for some reason reading it. I think because the plural is in the wrong place. Is the "s" a typo?

Author: Changed to `Transferring files to client`

- Working with Embedded JavaScript templating (ejs) - [examples/ejs](./ejs)

> - Example pages for HTTP error - [examples/error-pages](./error-pages)

Reviewer: This description also reads weird, including the part where this list of examples, I think "Example" is probably implied. What about something like "Creating error pages"?

- Working with error middleware - [examples/error](./error)
- Simple request handler - [examples/hello-world](./hello-world)
- Markdown as template engine - [examples/markdown](./markdown)
- Working with multiple express routers - [examples/multi-router](./multi-router)
- Accepting multipart-encoded forms - [examples/multipart](./multipart)
- MVC style controllers - [examples/mvc](./mvc)
- Tracking online user activity with `online` and `redis` packages - [examples/online](./online)
- Working with route parameters - [examples/params](./params)

> - Multiple HTTP operation on same resource - [examples/resource](./resource)

Reviewer: ```suggestion
- Multiple HTTP operations on the same resource - [examples/resource](./resource)
```

I am assuming the missing plural is a typo

- Organizing routes using map - [examples/route-map](./route-map)
- Working with route middleware - [examples/route-middleware](./route-middleware)
- Organizing routes per each resource - [examples/route-separation](./route-separation)

> - Redis-based search example - [examples/search](./search)

Reviewer: ```suggestion
- Search API - [examples/search](./search)
```

The example is less about Redis and more about how one would create a search using Express

> - Redis-based session - [examples/session](./session)

Reviewer: ```suggestion
- User sessions - [examples/session](./session)
```

The example is less about using Redis and more about sessions in general (which is why there are two different apps in there).

> - Serving static files example - [examples/static-files](./static-files)

Reviewer: ```suggestion
- Serving static files - [examples/static-files](./static-files)
```

- Working with virtual hosts - [examples/vhost](./vhost)
- Rendering views dynamically - [examples/view-constructor](./view-constructor)
- Saving data in request object between middleware calls - [examples/view-locals](./view-locals)

> - Simple API service example - [examples/web-service](./web-service)

Reviewer: ```suggestion
- Simple API service - [examples/web-service](./web-service)
```
